File: Android13/MyAndroidProjects/app/src/main/java/com/example/myandroidprojects/MainActivity.java
package com.example.myandroidprojects;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

/*  NOTE: inheritance chain -> AppCompatActivity -> FragmentActivity -> ComponentActivity -> androidx.core.app.ComponentActivity -> Activity
    Purpose: a single, focused thing that the user can do
    most activities interact with user,
    so Activity class takes care of creating a window for you
    to place your UI with setContentView
 */
public class MainActivity extends AppCompatActivity {

    // NOTE: called when activity is starting, where most initialization should go
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RecyclerView projectList = findViewById(R.id.recycler_view_projects);

        Project[] projects = {
                new Project("Getting Started App",
                        "Our very first project, the default 'Hello World' app!", R.drawable.getting_started),
                new Project("Motivational Quotes App",
                        "Making a simple change to the layout, with my own personal motivational quote for software developers", R.drawable.quote),
                new Project("BMI Calculator App",
                        "A real life working BMI calculator teaching me Variables, Methods, and Conditional Logic", R.drawable.calculator),
                new Project("Inches Converter",
                        "A basic converter to convert inches to meters (my own personal calculator app made on my own)", R.drawable.tape),
                new Project("The Hungry Developer",
                        "A menu app for a fictional restaurant, learning about Activities, Classes & Objects, Arrays, Intents, and ListViews", R.drawable.hungry_developer)
        };

        ProjectsAdapter adapter = new ProjectsAdapter(projects);
        projectList.setAdapter(adapter);
    }
}

File: DiceRoller/app/src/main/java/com/kryptopass/diceroller/MainActivity.kt
package com.kryptopass.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import android.widget.Toast
import com.kryptopass.diceroller.databinding.ActivityMainBinding

/**
 * A single, focused thing that the user can do.
 * Almost all activities interact with the user, so the Activity class takes
 * care of creating a window for you in which you can place your UI with.
 * AppCompatActivity is a subclass of Android that gives us access to modern
 * Android features while still supporting older versions of the Android OS.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var dieImage: ImageView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        dieImage = binding.imageViewDie
        setupButtonRoll()
    }
    private fun setupButtonRoll() {
        binding.btnRoll.setOnClickListener {
            rollDie()
        }
    }

    private fun rollDie() {
        val drawableResource = when ((1..6).random()) {
            1 -> R.drawable.die1
            2 -> R.drawable.die2
            3 -> R.drawable.die3
            4 -> R.drawable.die4
            5 -> R.drawable.die5
            else ->  R.drawable.die6
        }
        dieImage.setImageResource(drawableResource)
    }
}